src/hw/requireApps.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import { Observable, from, of, throwError, defer } from "rxjs";
+import { concatMap, catchError, delay } from "rxjs/operators";
+import {
+  TransportStatusError,
+  FirmwareOrAppUpdateRequired,
+  DisconnectedDeviceDuringOperation,
+  DisconnectedDevice,
+} from "@ledgerhq/errors";
+import type { DeviceModelId } from "@ledgerhq/devices";
+import type { DerivationMode } from "../types";
+import { withDevice } from "./deviceAccess";
+import { isDashboardName } from "./isDashboardName";
+import getAppAndVersion from "./getAppAndVersion";
+import type { ConnectAppEvent } from "./connectApp";
+import { attemptToQuitApp } from "./connectApp";
+import { streamAppInstall } from "../apps/hw";
+
+export type RequiresDerivation = {|
+  currencyId: string,
+  path: string,
+  derivationMode: DerivationMode,
+  forceFormat?: string,
+|};
+
+export type Input = {
+  modelId: DeviceModelId,
+  devicePath: string,
+  appNames: string[],
+};
+
+const cmd = ({ devicePath, appNames }: Input): Observable<ConnectAppEvent> =>
```

Comment:
I don't understand why this is not done in actions/app , yet you add to change code here.

i miss context of this PR to have any conclusion on this.

let's rediscuss next week.

Reply:
That was the whole point of wanting you to take a look. I can rework it so it's in the same connectApp but it seemed odd because the end result is not "opening" an app. 🤔